opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -151,82 +172,64 @@ impl OtlpEncoder {
                 fields.push((attr.key.clone().into(), type_id));
             }
         }
-        fields.sort_by(|a, b| a.0.cmp(&b.0)); // Sort fields by name consistent schema ID generation
-        fields
+
+        // Sort fields by name for consistent schema ID generation
+        fields.sort_by(|a, b| a.0.cmp(&b.0));
+
+        // Hash field names and types while converting to FieldDef
+        let field_defs: Vec<FieldDef> = fields
             .into_iter()
             .enumerate()
-            .map(|(i, (name, type_id))| FieldDef {
-                name,
-                type_id,
-                field_id: (i + 1) as u16,
+            .map(|(i, (name, type_id))| {
+                // Hash field name and type for schema ID
+                name.hash(&mut hasher);
+                type_id.hash(&mut hasher);
+
+                FieldDef {
+                    name,
+                    type_id,
+                    field_id: (i + 1) as u16,
+                }
             })
-            .collect()
-    }
-
-    /// Calculate schema ID from field specifications
-    fn calculate_schema_id(fields: &[FieldDef]) -> u64 {
-        use std::collections::hash_map::DefaultHasher;
-        use std::hash::{Hash, Hasher};
+            .collect();
 
-        let mut hasher = DefaultHasher::new();
-
-        for field in fields {
-            field.name.hash(&mut hasher);
-            field.type_id.hash(&mut hasher);
-        }
-
-        hasher.finish()
+        let schema_id = hasher.finish();
+        (field_defs, schema_id)
     }
 
-    /// Get or create schema with field ordering information
+    /// Get or create schema - fields are accessible via returned schema entry
     fn get_or_create_schema(
         &self,
         schema_id: u64,
         field_info: Vec<FieldDef>,
-    ) -> (CentralSchemaEntry, Vec<FieldDef>) {
+    ) -> CentralSchemaEntry {
         // Check cache first
-        if let Some((schema, schema_md5, field_info)) =
-            self.schema_cache.read().unwrap().get(&schema_id)
         {
-            return (
-                CentralSchemaEntry {
+            if let Some((schema, schema_md5)) = self.schema_cache.read().unwrap().get(&schema_id) {
+                return CentralSchemaEntry {
                     id: schema_id,
                     md5: *schema_md5,
                     schema: schema.clone(),
```

Comment:
The cloning operations on happy path seem expensive. `BondEncodedSchema` cloning would involve cloning the fied_info vec and the encoded bytes vec along with some Strings. We should be able to use `Arc`s to avoid these allocations.

Reply:
Done,  BondEncodedSchema now internally maintains the Arc to the encoded schema bytes, which makes the cloning cheap. Thanks for the suggestion.